src/net/udp/frame.rs

```diff
@@ -81,29 +82,32 @@ impl<C: UdpCodec> Sink for UdpFramed<C> {
     type SinkError = io::Error;
 
     fn start_send(&mut self, item: C::Out) -> StartSend<C::Out, io::Error> {
-        if self.wr.len() > 0 {
+        trace!("sending frame");
+
+        if !self.flushed {
             try!(self.poll_complete());
-            if self.wr.len() > 0 {
```

Comment:
I think this clause may still be needed? If `poll_complete` returns `NotReady` then we need to return `NotReady` here I think?